src/back/hlsl/writer.rs

```diff
@@ -126,7 +126,17 @@ impl<'a, W: fmt::Write> super::Writer<'a, W> {
                     crate::MathFunction::Asinh
                     | crate::MathFunction::Acosh
                     | crate::MathFunction::Atanh
-                    | crate::MathFunction::Unpack2x16float => {
+                    | crate::MathFunction::Unpack2x16float
+                    | crate::MathFunction::Unpack2x16snorm
+                    | crate::MathFunction::Unpack2x16unorm
+                    | crate::MathFunction::Unpack4x8snorm
+                    | crate::MathFunction::Unpack4x8unorm
+                    // TODO: These use multiple args, unsure how to bake them
```

Comment:
Do you mean for insert/extract bits? If so, you'd have to insert the other `arg`s in `need_bake_expressions` as well.

Reply:
I'm not sure why, but insert and extract bits both work without having their expressions baked.

Maybe because they're gated on arg1/2/3 being `Some()`?

Reply:
I guess it works fine but the chain of expressions will be duplicated (written multiple times).